src/Microsoft.ServiceFabric.Client.Http/ImageStoreClient.cs

```diff
@@ -34,128 +35,177 @@ public ImageStoreClient(ServiceFabricHttpClient httpClient)
-        public Task<ImageStoreContent> GetImageStoreRootContentAsync(
+        public async Task<ImageStoreContent> GetImageStoreRootContentAsync(
             long? serverTimeout = 60,
             CancellationToken cancellationToken = default(CancellationToken))
         {
-            serverTimeout?.ThrowIfOutOfInclusiveRange("serverTimeout", 1, 4294967295);
-            var requestId = Guid.NewGuid().ToString();
-            var url = "ImageStore";
-            var queryParams = new List<string>();
-            
-            // Append to queryParams if not null.
-            serverTimeout?.AddToQueryParameters(queryParams, $"timeout={serverTimeout}");
-            queryParams.Add("api-version=6.0");
-            url += "?" + string.Join("&", queryParams);
-            
-            HttpRequestMessage RequestFunc()
+
+            var store = await GetImageStoreConnectionString();
+            if (IsLocalStore(store))
             {
-                var request = new HttpRequestMessage()
-                {
-                    Method = HttpMethod.Get,
-                };
-                return request;
+                var directories = Directory.EnumerateDirectories(imageStorePath, "*", SearchOption.TopDirectoryOnly).Select(d => new FolderInfo(d, (long)Directory.EnumerateFiles(d).Count())).ToList();
+                var files = Directory.EnumerateFiles(imageStorePath, "*", SearchOption.TopDirectoryOnly).Select(f => new System.IO.FileInfo(f)).Select(f => new Common.FileInfo(f.Length.ToString(), new FileVersion(), f.LastWriteTimeUtc, f.FullName.Replace(imageStorePath, ""))).ToList();
+                return new ImageStoreContent(files, directories);
             }
+            else
+            {
+                serverTimeout?.ThrowIfOutOfInclusiveRange("serverTimeout", 1, 4294967295);
+                var requestId = Guid.NewGuid().ToString();
+                var url = "ImageStore";
+                var queryParams = new List<string>();
 
-            return this.httpClient.SendAsyncGetResponse(RequestFunc, url, ImageStoreContentConverter.Deserialize, requestId, cancellationToken);
+                // Append to queryParams if not null.
+                serverTimeout?.AddToQueryParameters(queryParams, $"timeout={serverTimeout}");
+                queryParams.Add("api-version=6.0");
+                url += "?" + string.Join("&", queryParams);
+
+                HttpRequestMessage RequestFunc()
+                {
+                    var request = new HttpRequestMessage()
+                    {
+                        Method = HttpMethod.Get,
+                    };
+                    return request;
+                }
+
+                return await this.httpClient.SendAsyncGetResponse(RequestFunc, url, ImageStoreContentConverter.Deserialize, requestId, cancellationToken);
+            }
         }
 
         /// <inheritdoc />
-        public Task CopyImageStoreContentAsync(
+        public async Task CopyImageStoreContentAsync(
```

Comment:
Directly returning Task is better

Reply:
Since this method now has to retrieve the connection string, it can no longer just return a task.